class.jetpack-gutenberg.php

```diff
@@ -603,13 +603,25 @@ public static function enqueue_block_editor_assets() {
 			plugins_url( $blocks_dir . '/', JETPACK__PLUGIN_FILE )
 		);
 
+		// default: both settings are always active on WPCOM.
+		$has_likes   = true;
+		$has_sharing = true;
+
+		$in_jetpack = ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? false : true;
+		if ( $in_jetpack && method_exists( 'Jetpack', 'is_module_active' ) ) {
+			$has_likes   = Jetpack::is_module_active( 'likes' );
+			$has_sharing = Jetpack::is_module_active( 'sharedaddy' );
+		}
+
 		wp_localize_script(
 			'jetpack-blocks-editor',
 			'Jetpack_Editor_Initial_State',
 			array(
 				'available_blocks' => self::get_availability(),
 				'jetpack'          => array( 'is_active' => Jetpack::is_active() ),
 				'siteFragment'     => $site_fragment,
+				'hasLikes'         => $has_likes,
+				'hasSharing'       => $has_sharing,
```

Comment:
We were hoping for `Jetpack_Editor_Initial_State` to be the exception rather than the rule to communicate information from the server to the client, i.e. to not really set an example for people to add more fields to :sweat_smile: Would it be possible to either

- Use a mechanism as provided by `@wordpress/data` to fetch information from the network on the client side, or
- Use our existing `available_blocks` mechanism?

My gut feeling tells me that the latter might be a good fit for this scenario: Since there already are distinct modules for both Likes and Shares, we could expose them both through separate `Jetpack_Gutenberg::set_extension_available()` calls on the server.

On the client side, we could then
- move each the Likes and Shares checkbox to a file of their own
- and use the [Slot/Fill](https://wordpress.org/gutenberg/handbook/designers-developers/developers/components/slot-fill/) technique to 
  - turn the `LikesAndSharesPanel` component into a `<Slot />` (akin to our [`< JetpackPluginSidebar />`](https://github.com/Automattic/jetpack/blob/c9dc14ac0b6af78c23400f3383e3ae442b48f7f7/extensions/shared/jetpack-plugin-sidebar.js))
  - and conditionally add each checkbox to it.

I think this could be reasonably straight-forward -- let me know if I can help you with implementing it.

Reply:
Ah okay. FYI @roccotripaldi since this implementation was your suggestion -- I guess it's not recommended.

I'll see if I can get it working as you suggest.

Reply:
ah sorry for the bad recommendation. lesson learned.

Reply:
This is a good point, sorry I didn't spot it in my earlier review.

I'll see about adding a comment to this block in another PR to clarify why it exists and that is discouraged as a general purpose mechanism.

Reply:
I attempted to address this in https://github.com/Automattic/jetpack/pull/11709/commits/7a0df200bb1587bcf809eb4f1f4635542bd9aae0, but now I cannot seem to get anything to render. Maybe Gutenberge doesn't like nested slot/fills. Any ideas here?

Reply:
Never mind. I figured it out!